Sources/Nuke/ImageTask.swift

```diff
@@ -152,26 +151,33 @@ public final class ImageTask: Hashable {
         self.isDataTask = isDataTask
         self.pipeline = pipeline
         self.onEvent = onEvent
-        // TODO: remove thread hop
-        self.task = Task { @ImagePipelineActor in
```

Comment:
`ImageTask` is created from a `nonisolated` method (by design), which previously required to spawn a `Task` to perform it. By delaying the task start until you call once of its `async` APIs such as `response: ImageResponse`, we can take advantage of the asynchronous context created by however invokes it. There is no need to spawn one of our own. This eliminates a thread hop.